RocketChat/Rocket.Chat.ReactNative
# [FIX] Add channels to team's flow using different navigators

## Proposed changes


Refactored the options inside NewMessageStackNavigator to maintain the modal behavior when creating a new room from RoomListView and implemented a right-to-left animation when creating a channel from AddTeamChannel.

`Working on tablets too`

## Issue(s)	


The transition behavior when open `SelectedUsersView` was different if compare with `SelectedUsersView` while creating a channel in RoomListView.

## How to test or reproduce

- Create a channel
- Click on the icon `channel`
- Go to Add Channel To Team view
- The behavior should look near to behavior in RoomListView.


## Screenshots

<table>
<thead>
<tr>
<th>Before</th>
<th>Then</th>
</thead>
</tr>
<tbody>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/47038980/119559338-e889c400-bd78-11eb-9422-df7557f8c2aa.gif" height="450" />
</td>
<td>
<img src="https://user-images.githubusercontent.com/47038980/119578243-8b9b0780-bd92-11eb-9db5-37bef20c190c.gif"  height="450"/>
</td>
</tr>
</tbody>
</table>

## Types of changes




## Checklist



## Further comments
